gulpfile.js

```diff
@@ -0,0 +1,233 @@
+const path = require('path');
+
+const gulp = require('gulp');
+const plugins = require('gulp-load-plugins')();
+const pump = require('pump');
+const shelljs = require('shelljs');
+
+const Hexo = require('hexo');
+
+const dirs = {
+    dist: 'dist',
+    src: 'src/sonarwhal-theme',
+    tmp: 'src/sonarwhal-theme-optimized'
+};
+const imageExtensions = 'png,jpg,svg,ico';
+
+// ---------------------------------------------------------------------
+// | Helper tasks                                                      |
+// ---------------------------------------------------------------------
+
+gulp.task('clean:before', (done) => {
+    shelljs.rm('-rf', dirs.dist);
+    shelljs.rm('-rf', dirs.tmp);
+
+    done();
+});
+
+gulp.task('clean:after', (done) => {
+    [
+        `${dirs.tmp}/source/core`,
+        `${dirs.tmp}/source/js`,
+        `${dirs.tmp}/source/components`,
+        `${dirs.tmp}/source/images`,
+        `${dirs.tmp}/static`
+    ].forEach((folder) => {
+        shelljs.rm('-rf', folder);
+    });
+
+    done();
+});
+
+gulp.task('build:hexo', (done) => {
+    const hexo = new Hexo(process.cwd(), {});
+
+    hexo.init()
+        .then(() => {
+            return hexo.call('clean');
+        })
+        .then(() => {
+            return hexo.call('generate');
+        })
+        .then(done);
+});
+
+gulp.task('watch:hexo', (done) => {
+    const hexo = new Hexo(process.cwd(), {});
+
+    hexo.init()
+        .then(() => {
+            return hexo.call('generate', { watch: true });
+        })
+        .then(done);
+});
+
+gulp.task('copy:theme', () => {
+    return gulp.src(`${dirs.src}/**/*`)
+        .pipe(plugins.changed(dirs.tmp))
+        .pipe(gulp.dest(dirs.tmp));
+});
+
+gulp.task('move:static', () => {
+    return gulp.src(`${dirs.tmp}/static/**/*`)
+        .pipe(plugins.debug())
+        .pipe(gulp.dest(`${dirs.tmp}/source/static`));
+});
+
+gulp.task('optimize:templates', (cb) => {
+    const handlebarsRegex = [[/\{\{#[^}]+\}\}/, /\{\{\/[^}]+\}\}/], [/\{\{#[^}]+\}\}\{\{#[^}]+\}\}/, /\{\{\/[^}]+\}\}\{\{\/[^}]+\}\}/], [/\{\{\w[^}]+\}\}/, /\{\{\/\w[^}]+\}\}/]];
```

Comment:
Do you have a link that explains this?

Reply:
Not really, copy paste from Edgeportal after lots of trial and error.